Assets/_Project/Scripts/UserInterface/Views/ViewManager.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using PolSl.UrbanHealthPath.UserInterface.Components;
+using PolSl.UrbanHealthPath.UserInterface.Components.List;
+using PolSl.UrbanHealthPath.UserInterface.Initializers;
+using PolSl.UrbanHealthPath.UserInterface.Interfaces;
+using PolSl.UrbanHealthPath.UserInterface.Popups;
+using UnityEngine;
+
+namespace PolSl.UrbanHealthPath.UserInterface.Views
+{
+    public class ViewManager : MonoBehaviour
+    {
+        public ViewType CurrentViewType { get; private set; }
+        public GameObject CurrentView { get; private set; }
+        public ViewType LastViewType { get; private set; }
+        public History History { get; private set; }
+
+        [Serializable] public struct View
+        {
+            [SerializeField] private ViewType type;
+            [SerializeField] private GameObject viewObject;
+
+            public ViewType GetViewType()
+            {
+                return type;
+            }
+
+            public GameObject GetViewObject()
+            {
+                return viewObject;
+            }
+        }
+
+        [SerializeField] private View[] viewsWithTypes;
+
+        private Dictionary<ViewType, GameObject> _views;
+
+        private static ViewManager _instance;
+
+        private void Awake()
+        {
+            if (_instance != null && _instance != this)
+            {
+                Destroy(this.gameObject);
+            }
+            else
+            {
+                _instance = this;
+            }
+        }
+
+        private void Start()
+        {
+            _instance.CurrentView = null;
+
+            _instance._views = new Dictionary<ViewType, GameObject>();
+
+            foreach (var view in viewsWithTypes)
```

Comment:
var is forbidden by our lord and savior

Reply:
Foreach with a complex type is an exception from this rule ;) I wouldn't consider this example complex but I would let it pass.